docs/_content/ontwikkelaars/algemeen/authenticatie-autorisatie.md

```diff
@@ -25,41 +25,104 @@ op de API met gebruikersnaam/wachtwoord bijvoorbeeld. Dit ligt bij de gemeente.
 dat de integriteit van het token in orde is, en past daarna de claims in de
-payload toe. Hierbij bevat het token zelf alle informatie en hoeft er niet
-met een (centrale) autorisatie-server gecommuniceerd worden.
+payload toe.
 
-JWT bestaat uit drie stukken - een header, payload en signature. Uitgevers van
-tokens bestaan bij de organisaties (gemeentes). De APIs moeten de shared
-secret kennen om de signature te kunnen controleren. Naar veiligheid toe is
-het verstandig om per organisatie (of zelfs applicatie binnen een organisatie)
-een secret in te stellen. Dit brengt wat beheer-overhead met zich mee.
+JWT bestaat uit drie stukken - een header, payload en signature. De APIs moeten
+de shared secret kennen om de signature te kunnen controleren. Naar veiligheid
+toe is het verstandig om per organisatie (of zelfs applicatie binnen een
+organisatie) een secret in te stellen. Dit brengt wat beheer-overhead met zich
+mee.
 
 JWT kunnen niet gerevoked worden - daarom is het wenselijk om een korte expiry
 toe te kennen (bijvoorbeeld 24h). Indien er zich security breaches voordoen,
 dan moet het betrokken shared secret veranderd worden.
 
+In het JWT nemen we in de payload op wat het `client_id` is van de consumer.
+Dit dient om uniek te identificeren welke _applicatie_ met de provider
+communiceert. Applicaties registreren zich out-of-band bij providers, waarbij
+ze een `client_id` en een `secret` toegekend krijgen. Dit kan verlopen via een
+applicatiebeheerder bijvoorbeeld - out-of-band betekent hierbij dat dit niet
+zonder menselijke controle kan gebeuren. Het secret wordt gebruikt om het JWT
+cryptografisch te ondertekeken, waardoor de provider garantie heeft dat de
+consumer ook echt is wie die beweert te zijn (via het `client_id`).
+
+Een autorisatiecomponent (AC) is verantwoordelijk voor het bijhouden van
+autorisaties voor een applicatie. Typisch zien we dit voor ons als een
+component die centraal (per omgeving zoals testing, acceptatie en productie)
+binnen de gemeente ingericht is. De component ontsluit zelf een API waarmee
+de autorisaties van consumers ingericht worden.
+
+Een `Applicatie` resource karakteriseert zich met de volgende attributen:
+
+* `client_ids`: een lijst van client IDs. Dezelfde applicatie zal typisch
+  tegen meerdere componenten praten (ZRC & ZTC bijvoorbeeld). Om lekken te
+  voorkomen en goede beveiligingsprincipes toe te passen, gaan we uit van 1
+  client ID per component waarmee gecommuniceerd wordt. De applicatie zelf is
+  echter 1 entiteit waarop de autorisaties geconfigureerd worden, dus we houden
+  alle `client_ids` bij waarmee deze zich identificeert. Client IDs dienen
+  uniek te zijn.
+
+* `label`: een beschrijvend label voor de applicatie, zodat duidelijk is om
+  welke applicatie binnen de organisatie het gaat. Bijvoorbeeld:
+  "Melding Openbare Ruimte (burger)"
+
+* `autorisaties`: een lijst van autorisaties per zaaktype. Het zaaktype wordt
```

Comment:
Het gaat hier zo expliciet over zaaktypes dat het lijkt alsof het AC exclusief voor het ZRC is bedoeld?

Of exclusief voor ZGW? Of groter?

Reply:
In eerste instantie exclusief voor ZGW. Een v2 van de API kan een generieker gegevensmodel hebben, maar dat wou ik (nog) niet binnen scope trekken.

Reply:
Ok, maar dan zijn Zaaktypes dus een optioneel dingetje voor het ZRC. Nu lijkt het alsof Zaaktypes een vast onderdeel zijn van autorisaties.

Reply:
Dat zijn ze nu ook. Je kan een 'blanket' autorisatie op alles geven via een boolean flag en dan hoef je `autorisaties` als geheel niet mee te geven (soort van superuser dingetje).

Bestaande suites autoriseren ook op het niveau van zaaktype

Reply:
Ja maar een DRC heeft toch geen zaaktypes nodig?